locksmith/src/utils/key.ts

```diff
@@ -0,0 +1,64 @@
+import { ethers } from 'ethers'
+import * as subscriptionOperations from './../operations/subscriptionOperations'
+
+const MAX_UINT = ethers.constants.MaxUint256.toString()
+
+export const getMembershipState = async ({
+  key,
+  tokenAddress,
+  network,
+  lockAddress,
+  tokenId,
+  owner,
+}: {
+  key: any
+  tokenAddress: string
+  network: number
+  lockAddress: string
+  tokenId: string
+  owner: string
+}) => {
+  const isERC20 = tokenAddress && tokenAddress !== ethers.constants.AddressZero
+
+  const isRenewable =
+    Number(key?.lock?.version) >= 11 && key?.expiration !== MAX_UINT && isERC20
+
+  const subscriptions =
+    await subscriptionOperations.getSubscriptionsForLockByOwner({
```

Comment:
I am confused here because this function name indicates you are retrieving _all_ subscriptions (can we call them keys!) but then you are passing a `tokenId` which indicates there will be just 1?

```js
const subscription = subscriptionOperations.getSubscriptionForLockByOwner({tokenId, lockAddress, owner, networ})
```